drivers/pcie/host/pcie.c

```diff
@@ -370,6 +370,94 @@ pcie_bdf_t pcie_bdf_lookup(pcie_id_t id)
 	return PCIE_BDF_NONE;
 }
 
+/* Forward declaration needed since scanning a device may reveal a bridge */
+static bool scan_bus(int bus, const struct pcie_scan_opt *opt);
+
+static bool scan_dev(int bus, int dev, const struct pcie_scan_opt *opt)
+{
+	for (int func = 0; func <= PCIE_MAX_FUNC; func++) {
+		pcie_bdf_t bdf = PCIE_BDF(bus, dev, func);
+		uint32_t secondary = 0;
+		uint32_t id, type;
+		bool do_cb;
+
+		id = pcie_conf_read(bdf, PCIE_CONF_ID);
+		if (!PCIE_ID_IS_VALID(id)) {
+			continue;
+		}
+
+		type = pcie_conf_read(bdf, PCIE_CONF_TYPE);
+		switch (PCIE_CONF_TYPE_GET(type)) {
+		case PCIE_CONF_TYPE_STANDARD:
+			do_cb = true;
+			break;
+		case PCIE_CONF_TYPE_PCI_BRIDGE:
+			if (opt->recursive) {
+				uint32_t num = pcie_conf_read(bdf,
+							      PCIE_BUS_NUMBER);
+				secondary = PCIE_BUS_SECONDARY_NUMBER(num);
+			}
+			__fallthrough;
+		case PCIE_CONF_TYPE_CARDBUS_BRIDGE:
```

Comment:
Also need `__fallthrough` here.

Reply:
I just removed the CARDBUS case, since we don't do anything special for it.

FWIW, if it's part of the official coding style to require __fallthrough even when there are no lines between the case statements, it seems pretty wasteful, especially if you have lots of case statement stacked up like that. It also doesn't add much value: the purpose of __fallthrough is to make the intent clear whenever there might be some doubt over it. With nothing in between two consecutive case statements (not even empty lines) there really isn't any doubt over the intent. Anyway, this is a topic independent of this PR.